clients/src/test/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslServerTest.java

```diff
@@ -55,17 +58,60 @@
         tmp.put(SaslConfigs.SASL_JAAS_CONFIG, new Password(jaasConfigText));
         CONFIGS = Collections.unmodifiableMap(tmp);
     }
+    static class OAuthBearerTokenMock implements OAuthBearerToken {
+        @Override
+        public String value() {
+            return null;
+        }
+
+        @Override
+        public Set<String> scope() {
+            return null;
+        }
+
+        @Override
+        public long lifetimeMs() {
+            return 0;
+        }
+
+        @Override
+        public String principalName() {
+            return null;
+        }
+
+        @Override
+        public Long startTimeMs() {
+            return null;
+        }
+    }
     private static final AuthenticateCallbackHandler LOGIN_CALLBACK_HANDLER;
     static {
         LOGIN_CALLBACK_HANDLER = new OAuthBearerUnsecuredLoginCallbackHandler();
         LOGIN_CALLBACK_HANDLER.configure(CONFIGS, OAuthBearerLoginModule.OAUTHBEARER_MECHANISM,
                 JaasContext.loadClientContext(CONFIGS).configurationEntries());
     }
     private static final AuthenticateCallbackHandler VALIDATOR_CALLBACK_HANDLER;
+    private static final AuthenticateCallbackHandler EXTENSIONS_VALIDATOR_CALLBACK_HANDLER;
     static {
         VALIDATOR_CALLBACK_HANDLER = new OAuthBearerUnsecuredValidatorCallbackHandler();
         VALIDATOR_CALLBACK_HANDLER.configure(CONFIGS, OAuthBearerLoginModule.OAUTHBEARER_MECHANISM,
                 JaasContext.loadClientContext(CONFIGS).configurationEntries());
+        EXTENSIONS_VALIDATOR_CALLBACK_HANDLER = new OAuthBearerUnsecuredValidatorCallbackHandler() {
```

Comment:
I'm wondering if `OAuthBearerUnsecuredValidatorCallbackHandler` should recognize all extensions as being valid.  This class is not meant for production use, and any extensions that are added in the JAAS config are meant to be there because they were explicitly put there for testing/development purposes, so it seems to make sense that it should just blindly mark everything as being valid.  Thoughts?

Reply:
Yes, it does make sense. Will change